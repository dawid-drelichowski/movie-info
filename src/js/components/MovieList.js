import React, {Component} from 'react'
import PropTypes from 'prop-types'
import Movie from 'components/Movie'

export default class MovieList extends Component {
  static propTypes = {
    movies: PropTypes.array
  }
  static defaultProps = {
    movies: []
  }
  render () {
    return <ul>
      {this.props.movies.map(movie => <Movie key={movie.id} data={movie}/>)}
    </ul>
  }
}
