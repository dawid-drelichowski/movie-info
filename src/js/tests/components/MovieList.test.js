import React from 'react'
import MovieList from 'components/MovieList'
import renderer from 'react-test-renderer'
import config from 'tests/mocks/config.json'
import search from 'tests/mocks/search.json'
import repository from 'tests/mocks/repository'

describe('Movie list component', () => {
  function getRenderedComponent (movies) {
    return renderer.create(
      <MovieList config={config} repository={repository} movies={movies} />
    )
  }

  it('should contain 1 movie', () => {
    expect(getRenderedComponent(search.results.slice(-1)).toJSON()).toMatchSnapshot()
  })

  it('should contain 4 movies', () => {
    expect(getRenderedComponent(search.results).toJSON()).toMatchSnapshot()
  })
})
