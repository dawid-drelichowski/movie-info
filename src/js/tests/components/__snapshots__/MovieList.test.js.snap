// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie list component should contain 1 movie 1`] = `
<div
  className="form-row"
>
  <ul
    className="col-12 list-group"
  >
    <li
      className="list-group-item"
      onClick={[Function]}
    >
      <strong>
        Minions: Puppy
      </strong>
      <img
        alt="Minions: Puppy"
        className="rounded img-thumbnail float-right"
        src="https://image.mock.api/small/maBSw4YgUzjQoowiMI51FG0Npzc.jpg"
      />
      
    </li>
  </ul>
</div>
`;

exports[`Movie list component should contain 4 movies 1`] = `
<div
  className="form-row"
>
  <ul
    className="col-12 list-group"
  >
    <li
      className="list-group-item"
      onClick={[Function]}
    >
      <strong>
        Minions
      </strong>
      <img
        alt="Minions"
        className="rounded img-thumbnail float-right"
        src="https://image.mock.api/small/q0R4crx2SehcEEQEkYObktdeFy.jpg"
      />
      
    </li>
    <li
      className="list-group-item"
      onClick={[Function]}
    >
      <strong>
        Minions: Competition
      </strong>
      <img
        alt="Minions: Competition"
        className="rounded img-thumbnail float-right"
        src="https://image.mock.api/small/d7wrcnERHpa6fA3rhvTllIitGRm.jpg"
      />
      
    </li>
    <li
      className="list-group-item"
      onClick={[Function]}
    >
      <strong>
        Mower Minions
      </strong>
      <img
        alt="Mower Minions"
        className="rounded img-thumbnail float-right"
        src="https://image.mock.api/small/Ao9clrCn3Ct4a92amHWzfnOCLCh.jpg"
      />
      
    </li>
    <li
      className="list-group-item"
      onClick={[Function]}
    >
      <strong>
        Minions: Puppy
      </strong>
      <img
        alt="Minions: Puppy"
        className="rounded img-thumbnail float-right"
        src="https://image.mock.api/small/maBSw4YgUzjQoowiMI51FG0Npzc.jpg"
      />
      
    </li>
  </ul>
</div>
`;
